src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -1084,9 +1084,13 @@ module Genesis_proof = struct
           "No genesis proof file was found for $base_hash and not allowed to \
            generate a new genesis proof"
           ~metadata:[("base_hash", Base_hash.to_yojson base_hash)] ;
-        Deferred.Or_error.errorf
-          "No genesis proof file was found and not allowed to generate a new \
-           genesis proof"
+        Deferred.Or_error.of_exn
+          (Mina_user_error.Mina_user_error
+             { where= Some "generating a genesis proof"
+             ; message=
+                 "Hint: pass the flag --generate-genesis-proof true. For \
+                  example,\n\
+                  mina.exe daemon --generate-genesis-proof true" })
```

Comment:
I think the executable is just `mina` in distributed containers.

Could you use argv[0] here?